### PR TITLE
Add TextFromSubject task to FEM lab

### PR DESCRIPTION
**Pull request by mcbouslog**
Staging branch URL: https://pr-6131.pfe-preview.zooniverse.org
- https://pr-6131.pfe-preview.zooniverse.org/lab/1952/workflows/3602

See related [FEM DigiLeap project](https://github.com/zooniverse/front-end-monorepo/projects/17) and specifically documentation related https://github.com/zooniverse/front-end-monorepo/pull/3008.

Describe your changes.
- add `textFromSubject` task to admin project experimental features
- add `Text from Subject` option to FEM lab task options
- add `textFromSubject` task with placeholder component
- add `textFromSubject` task editor

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `npm ci` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on GitHub Actions?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/tasks/textFromSubject/editor.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  workflow
+}) {
+  const handleChange = handleInputChange.bind(workflow);
+
+  return (
+    <div className="text-editor">
+      <section>
+        <div>
+          <AutoSave resource={workflow}>
+            <span className="form-label">Main text</span>
```
Comment (eatyourgreens) [inline app/classifier/tasks/textFromSubject/editor.jsx:21]:
```suggestion
            <label for={`${taskPrefix}-instruction`} className="form-label">Main text</label>
```

File: app/classifier/tasks/textFromSubject/editor.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            <br />
+            <textarea
```
Comment (eatyourgreens) [inline app/classifier/tasks/textFromSubject/editor.jsx:23]:
```suggestion
            <textarea
              id={`${taskPrefix}-instruction`}
```
All form inputs must be labelled.

**Comment by mcbouslog** [comment]
Thank you @eatyourgreens for the `label` comments! I updated accordingly. I'm going to tag @shaunanoordin for review because this touches the FEM Lab code a little, but of course don't hesitate to add any additional comments.

**Comment by shaunanoordin** [comment]
I'm taking a look at this now! Code read looks good, I'm now running some tests to see if I understand the changes correctly, and if there's anything to be aware of in terms of the FEM-Lab project. 👌

**Review by shaunanoordin — APPROVED** [review]
## PR Review

This PR allows Project Owners to add and edit "Text From Subjects" Tasks on the Project Builder > Workflow Editor.

- The TextFromSubjects Task editor...
  - only appears on the WF editor if the project has the experimental_tool enabled.
  - appears correctly on the FEM Lab. (Note: it does NOT appear on classic PFE Lab)
  - has its own compartmentalised textFromSubject/editor.jsx file _(good gravy, thank you for not overloading the generic task editor further! 👍 )_

As far as the FEM-lab is concerned, this is what I was looking for:
- Changes to lab-fem/workflow.jsx is kept to a minimum
- The editor code is compartmentalised into its own file (i.e. it's kept modular, so it should work with the newer FEM-lab updates)

Code read looks good, and tests look good on localhost with macOS + Chrome. (I was able to add & edit a TextFromSubject Task to my own test project on the Project Builder, albeit without any valid text Subjects.)

### Status

LGTM! 👍

**Pull request merged**